src/Auth/GenericAuthenticator.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\Auth;
+
+use Tempest\Http\Session\Session;
+
+abstract class GenericAuthenticator implements Authenticator
```

Comment:
Let's refactor to a trait `AuthenticatorWithSession` or something alike

Reply:
Thanks! I’ll do it!